substrate/frame/staking-async/src/tests/slashing.rs

```diff
@@ -262,12 +264,17 @@ fn subsequent_reports_in_same_span_pay_out_less() {
 		);
 		Session::roll_next();
 
-		let prior_payout = reward;
-		// F1 * (reward_proportion * slash - prior_payout)
-		// 50% * (10% * (initial_balance / 2) - prior_payout)
-		let reward = ((initial_balance / 20) - prior_payout) / 2;
-		assert_eq!(reward, 20);
-		assert_eq!(asset::total_balance::<T>(&1), initial_balance_1 + prior_payout + reward);
+		let prior_slash = slash;
+		let prior_reward = reward;
+
+		// since the slash is in the same era, the prior slash is discounted.
+		// total slash is 1000/2 = 500, out of which 200 is already slashed. So net slash is 300.
```

Comment:
Nice, I like this test! can you also assert on the `Slash` related events after each step? I think it would help verify even more.